pkg/safeguards/preprocessing_helpers.go

```diff
@@ -0,0 +1,108 @@
+package safeguards
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"gopkg.in/yaml.v2"
+	"helm.sh/helm/v3/pkg/chart"
+	"helm.sh/helm/v3/pkg/chart/loader"
+	"helm.sh/helm/v3/pkg/chartutil"
+	"helm.sh/helm/v3/pkg/engine"
+)
+
+// Given a Helm chart directory or file, renders all templates and writes them to the specified directory
+func RenderHelmChart(isFile bool, mainChartPath, tempDir string) error {
+	if isFile { // Get the directory that the Chart.yaml lives in
+		mainChartPath = filepath.Dir(mainChartPath)
+	}
+
+	mainChart, err := loader.Load(mainChartPath)
+	if err != nil {
+		return fmt.Errorf("failed to load main chart: %s", err)
+	}
+
+	loadedCharts := make(map[string]*chart.Chart) // map of chart path to chart object
+	loadedCharts[mainChartPath] = mainChart
+
+	// Load subcharts and dependencies
+	for _, dep := range mainChart.Metadata.Dependencies {
+		// Resolve the chart path based on the main chart's directory
+		chartPath := filepath.Join(mainChartPath, dep.Repository[len("file://"):])
+		chartPath = filepath.Clean(chartPath)
+
+		subChart, err := loader.Load(chartPath)
+		if err != nil {
+			return fmt.Errorf("failed to load chart: %s", err)
+		}
+		loadedCharts[chartPath] = subChart
+	}
+
+	for chartPath, chart := range loadedCharts {
+		valuesPath := filepath.Join(chartPath, "values.yaml") // Enforce that values.yaml must be at same level as Chart.yaml
+		mergedValues, err := getValues(chart, valuesPath)
+		if err != nil {
+			return fmt.Errorf("failed to load values: %s", err)
+		}
+		e := engine.Engine{Strict: true}
+		renderedFiles, err := e.Render(chart, mergedValues)
+		if err != nil {
+			return fmt.Errorf("failed to render chart: %s", err)
+		}
+
+		// Write each rendered file to the output directory with the same name as in templates/
+		for filePath, content := range renderedFiles {
+			outputFilePath := filepath.Join(tempDir, filepath.Base(filePath))
+			if err := os.WriteFile(outputFilePath, []byte(content), 0644); err != nil {
```

Comment:
copying discussion here:
we can likely bypass writing to the temp directory and instead validate the manifests directly in memory since we are just reading them back out from the temp directory after

Reply:
Changed return type to include array of MaifestFile struct